PEPit/functions/smooth_convex_function.py

```diff
@@ -46,3 +47,8 @@ def __init__(self,
                          reuse_gradient=True,
                          mu=0,
                          L=L)
+
+        if self.L == np.inf:
+            print("\033[96m(PEPit) The class of smooth convex functions is necessarily differentiable.\n"
+                  "When setting L to infinity, please use the class of convex functions (ConvexFunction),\n "
```

Comment:
"Instead of setting L to infinity, ..."

Ou "to instantiate a convex function, please avoid using the class SmoothConvexFunction with L==infinity. Instead, please use the class ConvexFunction".